zppy/templates/pcmdi_diags.bash

```diff
@@ -0,0 +1,1544 @@
+#!/bin/bash
+{% include 'slurm_header.sh' %}
+
+{{ environment_commands }}
+
+# Turn on debug output if needed
+debug={{ debug }}
+if [[ "${debug,,}" == "true" ]]; then
+  set -x
+fi
+
+# Need this setup as otherwise can not generate diagnostics
+export UCX_SHM_DEVICES=all # or not set UCX_NET_DEVICES at all
+
+# Make sure UVCDAT doesn't prompt us about anonymous logging
+export UVCDAT_ANONYMOUS_LOG=False
+
+# Script dir
+cd {{ scriptDir }}
+
+# Get jobid
+id=${SLURM_JOBID}
+
+# Update status file
+STARTTIME=$(date +%s)
+echo "RUNNING ${id}" > {{ prefix }}.status
+
+# Basic definitions
+case="{{ case }}"
+www="{{ www }}"
+y1={{ year1 }}
+y2={{ year2 }}
+Y1="{{ '%04d' % (year1) }}"
+Y2="{{ '%04d' % (year2) }}"
+{% if run_type == "model_vs_model" %}
+ref_Y1="{{ '%04d' % (ref_year1) }}"
+ref_Y2="{{ '%04d' % (ref_year2) }}"
+{%- endif %}
+run_type="{{ run_type }}"
+tag="{{ tag }}"
+
+results_dir=${tag}_${Y1}-${Y2}
+
+ref_name={{ ref_name }}
+
+#info for pcmdi specific data structure
+case_id=v$(date '+%Y%m%d')
+
+# Create temporary workdir
+workdir=`mktemp -d tmp.${id}.XXXX`
+cd ${workdir}
+
+# Create results directory
+if [ ! -d ${results_dir} ];then
+  mkdir -p ${results_dir}
+fi
+#directory to save land/sea mask generated by pcmdi
+fixed_dir="${results_dir}/fixed"
+if [ ! -d ${fixed_dir} ];then
+  mkdir -p ${fixed_dir}
+fi
+
+# Prepare data files for pcmdi diagnostics, which is achieved by two steps:
+# (a) convert e3sm output to cmip type, which used the "e3sm_to_cmip" function
+#     available at zppy (modifications are made to process more variables and
+#     3D fileds at fixed pressure levels).
+# (b) locate observations in e3sm diagnostics and convert them to the pcmdi preferred
+#     data format
+#file to specify reference data used to derive the diagnostic metrics
+cat > reference_alias.json << EOF
+{% include reference_alias %}
+EOF
+#regions specified to derive global/regional mean metrics
+cat > regions_specs.json << EOF
+{% include regions_specs %}
+EOF
+#file include derived variables
+cat > derived_variable.json << EOF
+{% include derived_variable %}
+EOF
+#file to genereate land/sea mask data if not available
+cat > process_sftlf.py << EOF
+{% include process_sftlf %}
+EOF
+
+{%- if ("mean_climate" in sets) %}
+#file to genereate figures for mean climate metrics(temporary)
+cat > mean_climate_plot_parser.py << EOF
+{% include clim_plot_parser %}
+EOF
+#file to genereate figures for mean climate metrics(temporary)
+cat > mean_climate_plot_driver.py << EOF
+{% include clim_plot_driver %}
+EOF
+{%- endif %}
+
+# script for pcmdi pre-processing
+cat > collect_data.py << EOF
```

Comment:
I'm thinking this `.py` file should be moved over to zppy-interfaces. The goal is for the `.bash` file to be extremely light. Ideally, it's zppy calling some other package (e.g., `e3sm_diags` or `zi-global-time-series`, which is the global-time-series command in zppy-interfaces) with the proper arguments, and that's pretty much it.